### PR TITLE
iOS V10 PointAnnotation Drag and Selection Support

### PR DESCRIPTION
**Pull request by mysport12**
## Description

- Added draggable support for PointAnnotation
- Added support for onDrag, onDragEnd, onDragStart, onDeselected, and onSelected

## Checklist

- [x] I have tested this on a device/simulator for each compatible OS
- [ ] I updated the documentation with running `yarn generate` in the root folder
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typings files (`index.d.ts`)
- [ ] I added/ updated a sample (`/example`)

### CODE REVIEW

File: android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/modules/RCTMGLOfflineModule.kt [hunk 1]
```diff
@@ -378,6 +378,24 @@ class RCTMGLOfflineModule(private val mReactContext: ReactApplicationContext) :
...
+        val sourcePath = Paths.get(mReactContext.filesDir.absolutePath + "/mbgl-offline.db")
+        val targetPath = Paths.get(targetPathName + "/map_data.db")
+
+        try {
+            val directory = File(targetPathName)
+            directory.mkdirs()
+            Files.move(sourcePath, targetPath, StandardCopyOption.REPLACE_EXISTING)
+            Log.d("TAG","v10 cache directory created successfully")
+        } catch (e: Exception) {
+            Log.d("TAG", "${e}... file move unsuccessful")
```
Comment (mfazekas) [inline android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/modules/RCTMGLOfflineModule.kt:395]:
```suggestion
            Log.e("RCTMGLOfflineModule", "${e}... file move unsuccessful")
```

File: android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/modules/RCTMGLOfflineModule.kt [hunk 2]
```diff
@@ -378,6 +378,24 @@ class RCTMGLOfflineModule(private val mReactContext: ReactApplicationContext) :
...
+        // Old and new cache file paths
+        val targetPathName = mReactContext.filesDir.absolutePath + "/.mapbox/map_data"
+        val sourcePath = Paths.get(mReactContext.filesDir.absolutePath + "/mbgl-offline.db")
+        val targetPath = Paths.get(targetPathName + "/map_data.db")
+
+        try {
+            val directory = File(targetPathName)
+            directory.mkdirs()
+            Files.move(sourcePath, targetPath, StandardCopyOption.REPLACE_EXISTING)
+            Log.d("TAG","v10 cache directory created successfully")
```
Comment (mfazekas) [inline android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/modules/RCTMGLOfflineModule.kt:393]:
```suggestion
```

I think this can be removed

**Review by mfazekas — COMMENTED** [review]
Thanks, looks awesome!

File: ios/RCTMGL-v10/RCTMGLPointAnnotation.swift [hunk 3]
```diff
@@ -10,12 +10,20 @@ final class WeakRef<T: AnyObject> {
 }
 
 
-class RCTMGLPointAnnotation : UIView, RCTMGLMapComponent {
+class RCTMGLPointAnnotation : RCTMGLSource {
```
Comment (mfazekas) [inline ios/RCTMGL-v10/RCTMGLPointAnnotation.swift:13]: PointAnnotation is not a source, we can extract some common subclass but inheriting from `RCTMGLSource` sounds fishy for me.

**Review by mfazekas — CHANGES_REQUESTED** [review]

File: ios/RCTMGL-v10/RCTMGLPointAnnotation.swift (hunk 3 above)
Comment (mysport12) [reply ios/RCTMGL-v10/RCTMGLPointAnnotation.swift:13]: Sorry I haven't responded to this yet. We needed to jump back to v8 and I haven't had a chance to look into this further. If I can this week I will.

Comment (mysport12) [reply ios/RCTMGL-v10/RCTMGLPointAnnotation.swift]: @mfazekas attempted to break this out to a common subclass per your request. I am not the most proficient at Swift so there may be a cleaner/better way to go about it. Let me know your thoughts.

**Review by mfazekas — APPROVED** [review]

**Pull request merged**